docs/interfaces.rst

```diff
@@ -0,0 +1,105 @@
+Interface types
+===============
+
+Just like previously described :ref:`unions <unions>`, interfaces enable you to create fields that resolve to a value that is one of a few possible types.
```

Comment:
Should we really describe interfaces in comparison to unions? I know that from library implementation point of view they share some implementation details, but idea is wildly different.
I would start with defining interface as an abstract type, similar to those available in other languages, that enforces on types implementing it existence of defined fields and enable polymorphic behaviors in API. For example, one might return list of objects that shares common features.

Reply:
I agree. Care to create a change suggestion?